projects/doubleup/index.js

```diff
@@ -23,13 +27,28 @@ async function tvl(api) {
     const housePool = house?.fields?.house_pool;
     const houseTvl = Number(housePipeDebt) + Number(housePool);
 
-    api.add(coinType, houseTvl);
-
     const pipeDebt = house?.fields?.pipe_debt?.fields?.value;
     const _tvl = Number(pipeDebt) + Number(house?.fields?.pool);
 
-    api.add(coinType, _tvl);
+    if(coinType === UPUSD || coinType === "0x49f123d62df1db5a735463f3817dcd53aa084173cbe4593db6b30647b9801cb8::unihouse::FeeTag, 0x49f123d62df1db5a735463f3817dcd53aa084173cbe4593db6b30647b9801cb8::unihouse::HouseFeeConfig"){
+      continue;
+    }else {
+      console.log(`${coinType}`,houseTvl + _tvl);
+
+      api.add(coinType, houseTvl+_tvl);
+    }
+    
   }
+
+  // UPUSD
+  const upusd_factory = await sui.getObject(UPUSD_FACTORY)
+
+  const upusd_tvl = upusd_factory?.fields?.basic_supply?.fields?.supply;
+  
+  console.log(`${UPUSD}`,upusd_tvl);
+
+  api.add("0xdba34672e30cb065b1f93e3ab55318768fd6fef66c15942c9f7cb846e2f900e7::usdc::USDC", upusd_tvl);
```

Comment:
Where is this product on the doubleup UI? How does UPUSD maintain peg?

Reply:
Hi @waynebruce0x 

UPUSD is in our swap UI and we are backed 1-1 with USDC.
<img width="756" height="473" alt="image" src="https://github.com/user-attachments/assets/2f270039-fb88-4e93-b16d-d70e2e9469a3" />
At the bottom right corner


<img width="1499" height="945" alt="image" src="https://github.com/user-attachments/assets/f719db15-846f-41df-aa2a-27ea1240a345" />


If you have any concern, feel free to check the package we doing swap for UPUSD

this is the package Id: [0x5dc5ef7bcbcf4031b7012e58b6fa4f7fa4643e26ba25f54365da2ad73e845f15](https://suivision.xyz/package/0x5dc5ef7bcbcf4031b7012e58b6fa4f7fa4643e26ba25f54365da2ad73e845f15?tab=Code)

Reply:
We leave the USDC in the smart contract
<img width="1433" height="580" alt="image" src="https://github.com/user-attachments/assets/a33a70eb-fe8c-4dc1-b748-90a3e3da9f59" />

Reply:
Okay can we please count the balance of USDC in this contract instead of the supply of UPUSD?